src/main/java/org/scijava/io/DataHandle.java

```diff
@@ -74,43 +82,86 @@
 	void setLength(long length) throws IOException;
 
 	/**
+	 * Verifies that the handle has sufficient bytes available to read, returning
+	 * the actual number of bytes which will be possible to read, which might
+	 * be less than the requested value.
+	 * 
+	 * @param count Number of bytes to read.
+	 * @return The actual number of bytes available to be read.
+	 * @throws IOException If something goes wrong with the check.
+	 */
+	default long available(final long count) throws IOException {
+		final long remain = length() - offset();
+		return remain < count ? remain : count;
+	}
+
+	/**
+	 * Ensures that the handle has sufficient bytes available to read.
+	 * 
+	 * @param count Number of bytes to read.
+	 * @see #available(long)
+	 * @throws EOFException If there are insufficient bytes available.
+	 * @throws IOException If something goes wrong with the check.
+	 */
+	default void ensureReadable(final long count) throws IOException {
+		if (available(count) < count) throw new EOFException();
+	}
+
+	/**
 	 * Ensures that the handle has the correct length to be written to and extends
 	 * it as required.
 	 * 
-	 * @param writeLength Number of bytes to write.
+	 * @param count Number of bytes to write.
 	 * @return {@code true} if the handle's length was sufficient, or
 	 *         {@code false} if the handle's length required an extension.
-	 * @throws IOException If there is an error changing the handle's length.
+	 * @throws IOException If something goes wrong with the check, or there is an
+	 *           error changing the handle's length.
 	 */
-	default boolean validateLength(final int writeLength) throws IOException {
-		if (offset() + writeLength > length()) {
-			setLength(offset() + writeLength);
+	default boolean ensureWritable(final long count) throws IOException {
+		final long minLength = offset() + count;
+		if (length() < minLength) {
+			setLength(minLength);
 			return false;
 		}
 		return true;
 	}
 
+	/** Returns the current order of the stream. */
+	ByteOrder getOrder();
```

Comment:
Oh I see, the diff is just funny due to other inserted code and/or moved code blocks. That can be avoided by doing more granular commits.